### PR TITLE
fix(chore): added missing behaviors and updated state module

### PR DESCRIPTION
**Pull request by lubber-de**
## Description

- The hidden but always included state module was never documented, so i finally did this
- Added missing behaviors and typo to api and form

I basically took, fixed and enhanced the examples from https://github.com/Semantic-Org/Semantic-UI/issues/609#issuecomment-42430071

## Screenshots
image
image
image
image
image
image
image
image
image

### CODE REVIEW

**Review by ko2in — CHANGES_REQUESTED** [review]
Need consistent present tense words (Allow, Disallow, Set) with 's' like the others.

**Comment by lubber-de** [comment]
> Need consistent present tense words (Allow, Disallow, Set) with 's' like the others.

Done by 21ce1e61727d27d4d9e9afba6c638beee0ff1915

**Review by ko2in — CHANGES_REQUESTED** [review]

**Review by ko2in — CHANGES_REQUESTED** [review]
`a optional` should be `an optional` at line 214.

File: server/documents/behaviors/state.html.eco [hunk 1]
```diff
@@ -55,92 +126,100 @@ type        : 'UI Behavior'
...
-          <td>read cached response(url)</td>
-          <td>Reads a locally cached response for a URL</td>
+          <td>toggle state</td>
+          <td>Toggles between active/inactive state but checks if element is active and uses a possible attached <a href="/behaviors/api.html">api</a></td>
         </tr>
         <tr>
-          <td>write cached response(url, response)</td>
-          <td>Writes a cached response for a URL</td>
+          <td>flash text(text, duration, callback)</td>
+          <td>Changes the elements text for a optional given amount of time. After that, reverts the text to the previous value and optionally calls a given callback function</td>
```
Comment (ko2in) [inline server/documents/behaviors/state.html.eco:214]: `a optional` should be `an optional`

**Review by ko2in — CHANGES_REQUESTED** [review]

File: server/documents/behaviors/state.html.eco [hunk 2]
```diff
@@ -7,39 +7,110 @@ element     : 'State'
...
-      </div>
-      <div class="ui hidden divider"></div>
-      <div class="ui primary button">Click Me</div>
-      <div class="ui disabled button">Disabled</div>
+    <div class="no example">
+      <h2 class="ui dividing header">Description</h2>
+      <p>State allows you to do things like
+        <ul class="ui large list">
+            <li>Tie loading state to AJAX requests</li>
+            <li>Tie text to different states, so for example having a button go from 'Follow to Followed to Unfollow' on hover depending on its current state</li>
```
[the remark this answers is not in the document]
Comment (lubber-de) [reply server/documents/behaviors/state.html.eco:26]: No, "goes" is wrong as "go" is used together with "having" here.
"going" would be correct / usable as well, but as the original sentence came from the SUI authot (US citizen) anyway, i would just keep it as it is

however, i checked via
https://www.reverso.net/spell-checker/english-spelling-grammar/ and the sentence seems to be correct

File: server/documents/behaviors/state.html.eco (hunk 1 above)
Comment (lubber-de) [reply server/documents/behaviors/state.html.eco]: Fixed by 9e842ba0ce41424054d0922a31f8f40679a6468a

File: server/documents/behaviors/state.html.eco (hunk 2 above)
[the remark this answers is not in the document]
Comment (ko2in) [reply server/documents/behaviors/state.html.eco:26]: I see.

**Comment by ko2in** [comment]
LGTM :+1:

**Pull request merged**